Code splitting

sourabhmodi20051 opened the pull request:
Configure Redux with dynamic reducer injection
Added sample usage of redux in app using hooks

nimish-gupta commented:
Why are you using redux?

sourabhmodi20051 commented:
> Why are you using redux?

Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc.  It will be easy to maintain this state using redux.

nimish-gupta commented:
> > Why are you using redux?
> 
> Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.

React context?

sourabhmodi20051 commented:
> > > Why are you using redux?
> > 
> > 
> > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> 
> React context?

The context will share state in the tree, in messaging case we will have to share state with sibling components as well.

nimish-gupta commented:
F

> > > > Why are you using redux?
> > > 
> > > 
> > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > 
> > 
> > React context?
> 
> The context will share state in the tree, in messaging case we will have to share state with sibling components as well.

For sibling components, you can pass the state to the parent component. You want to share the state into tree, right? 
For sibling components you can pass state to them individually.

sourabhmodi20051 commented:
> F
> 
> > > > > Why are you using redux?
> > > > 
> > > > 
> > > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > > 
> > > 
> > > React context?
> > 
> > 
> > The context will share state in the tree, in messaging case we will have to share state with sibling components as well.
> 
> For sibling components, you can pass the state to the parent component. You want to share the state into tree, right?
> For sibling components you can pass state to them individually.

Will that not increase the code complexity? With hooks, all the heavy lifting of redux need not be done by us.

nimish-gupta commented:
> > F
> > > > > > Why are you using redux?
> > > > > 
> > > > > 
> > > > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > > > 
> > > > 
> > > > React context?
> > > 
> > > 
> > > The context will share state in the tree, in messaging case we will have to share state with sibling components as well.
> > 
> > 
> > For sibling components, you can pass the state to the parent component. You want to share the state into tree, right?
> > For sibling components you can pass state to them individually.
> 
> Will that not increase the code complexity? With hooks, all the heavy lifting of redux need not be done by us.

As if now I can't comment on the code complexity. Hooks has also made the use of context api

sourabhmodi20051 commented:
> > > F
> > > > > > > Why are you using redux?
> > > > > > 
> > > > > > 
> > > > > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > > > > 
> > > > > 
> > > > > React context?
> > > > 
> > > > 
> > > > The context will share state in the tree, in messaging case we will have to share state with sibling components as well.
> > > 
> > > 
> > > For sibling components, you can pass the state to the parent component. You want to share the state into tree, right?
> > > For sibling components you can pass state to them individually.
> > 
> > 
> > Will that not increase the code complexity? With hooks, all the heavy lifting of redux need not be done by us.
> 
> As if now I can't comment on the code complexity. Hooks has also made the use of context api

I went through Context API properly, I think redux is not required,  removed redux and added react context :)

sourabhmodi20051 commented on client/src/containers/UserStatus/UserStatus.js:
```diff
@@ -0,0 +1,22 @@
+import React, { useContext } from "react";
+import { AppContext } from "../App/AppContext";
+
+function UserStatus() {
```
Added Tests
